test/next/nu_module_example.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+import 'package:nuvigator/next.dart';
+import 'package:nuvigator/src/next/v1/annotations.dart';
+
+part 'nu_module_example.g.dart';
+
+class FriendRequestArgs {
+  int numberOfRequests;
+  double precision;
+  String name;
+  int age;
+}
+
+@NuModuleParser()
+class FriendRequestRouteExample
+    extends NuRoute<NuModule, FriendRequestArgs, void> {
+  @override
+  String get path => 'friend-requests';
+
+  @override
+  Future<bool> init(BuildContext context) {
+    return Future.delayed(const Duration(seconds: 3), () => true);
+  }
+
+  @override
+  ScreenType get screenType => materialScreenType;
+
+  @override
+  Widget build(BuildContext context, NuRouteMatch<FriendRequestArgs> match) {
+    return Nuvigator(
+      module: null,
+      screenType: materialScreenType,
+    );
+  }
+
+  @override
+  FriendRequestArgs parseParameters(Map<String, dynamic> map) =>
+      _$parseParameters(map);
```

Comment:
I am thinking now, WDYT about making this a getter that returns a function?
This would improve a bit the override because we would not need to explicitly declare the parameters and pass it forward, ie:
```dart
ParamsParser<FriendRequestArgs> get parseParameters => _$parseParameters;
```

Reply:
I like it, for me it appears more declarative and easy to understand.

Reply:
I will try the change to see how it behaves.

Reply:
I will try the change to see how it behaves.